code/modules/games/cards.dm

```diff
@@ -337,57 +638,186 @@
 	attack_verb = D.card_attack_verb
 	resistance_flags = D.card_resistance_flags
 
+/// Update our own card values to reflect that of some source hand
+/obj/item/cardhand/proc/update_values_from_cards(obj/item/cardhand/source)
+	if(isnull(source) || !istype(source))
+		return
+
+	hitsound = source.hitsound
+	force = source.force
+	throwforce = source.throwforce
+	throw_speed = source.throw_speed
+	throw_range = source.throw_range
+	attack_verb = source.attack_verb
+	resistance_flags = source.resistance_flags
+
 /obj/item/cardhand/attackby(obj/O, mob/user)
+	// Augh I really don't like this
 	if(length(cards) == 1 && is_pen(O))
 		var/datum/playingcard/P = cards[1]
 		if(P.name != "Blank Card")
-			to_chat(user,"<span class='notice'>You cannot write on that card.</span>")
+			to_chat(user,"<span class='notice'>You cannot write on this card.</span>")
 			return
-		var/t = rename_interactive(user, P, use_prefix = FALSE, actually_rename = FALSE)
-		if(t && P.name == "Blank Card")
-			P.name = t
+		var/card_text = rename_interactive(user, O, use_prefix = FALSE, actually_rename = FALSE)
+		if(card_text && P.name == "Blank Card")
+			P.name = card_text
 		// SNOWFLAKE FOR CAG, REMOVE IF OTHER CARDS ARE ADDED THAT USE THIS.
 		P.card_icon = "cag_white_card"
 		update_appearance(UPDATE_NAME|UPDATE_DESC|UPDATE_OVERLAYS)
-	else if(istype(O,/obj/item/cardhand))
+	else if(istype(O, /obj/item/cardhand))
 		var/obj/item/cardhand/H = O
-		if(H.parentdeck == parentdeck)
-			H.concealed = concealed
-			cards.Add(H.cards)
-			qdel(H)
-			update_appearance(UPDATE_NAME|UPDATE_DESC|UPDATE_OVERLAYS)
+		// if you're attacking a cardhand with one in hand, merge it into our deck.
+		// remember that "we" are the one being attacked here.
+		if(H.parent_deck_id == parent_deck_id)
+			if(!Adjacent(user))
+				return
+			var/datum/playingcard/chosen = select_card_radial(user)
+			if(!chosen)
+				return
+			user.visible_message(
+				"<span class='notice'>[user] adds [concealed ? "a card" : chosen.name] to [user.p_their()] hand.</span>",
+				"<span clas='notice'>You add [chosen.name] to your deck.</span>",
+				"<span class='notice'>You hear cards being shuffled together.</span>"
+			)
+			H.transfer_card_to_self(src, chosen)
 			return
 		else
 			to_chat(user, "<span class='notice'>You cannot mix cards from other decks!</span>")
 			return
-	..()
+	return ..()
 
 /obj/item/cardhand/attack_self(mob/user)
 	if(length(cards) == 1)
 		turn_hand(user)
 		return
-	user.set_machine(src)
-	interact(user)
+	var/card = select_card_radial(user)
+	if(isnull(card))
```

Comment:
lacks post selection adjaency check, also i recommend using QDELETED instead of null for cases where the item will get queued for deletion(but not immediately deleted due to reference in select_card)

Reply:
Addressed, using qdeleted now